### PR TITLE
All species can now wag their tails if they get one.

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

This refactors tail wagging, so it's based on the tail organ rather than specific species. Mostly.

The result is that now _all_ species can *wag, if they have a tail.

## Why It's Good For The Game

What good is a tail if you can't wag it?

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

23-05-16-1684211128-dreamseeker

</details>

## Changelog
:cl:
refactor: Refactored tail wagging code a bit.
tweak: All species can now *wag their tail if they have one.
/:cl:

### CODE REVIEW

File: code/datums/mood_events/generic_negative_events.dm [hunk 1]
```diff
@@ -96,9 +96,11 @@
 	var/mob/living/T = L.parent
 	if(ishuman(T))
 		var/mob/living/carbon/human/H = T
-		if(iscatperson(H))
-			H.dna.species.start_wagging_tail(H)
-			addtimer(CALLBACK(H.dna.species, TYPE_PROC_REF(/datum/species, stop_wagging_tail), H), 30)
+		if(iscatperson(H) || (istype(H.getorganslot(ORGAN_SLOT_EARS), /obj/item/organ/ears/cat) && istype(H.getorganslot(ORGAN_SLOT_TAIL), /obj/item/organ/tail/cat)))
```
Comment (imsynthex) [inline code/datums/mood_events/generic_negative_events.dm:99]: Keeping `iscatperson(H)` seems redundant(?)

**Comment by Tsar-Salat** [comment]
ERP powercreep.

Organ modularization is always good, speedmerge 😎

File: code/datums/mood_events/generic_negative_events.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/datums/mood_events/generic_negative_events.dm:99]: Agreed. Replace all species checks relating to emotes with just organ checks.

Or maybe, a `to_chat span class='warning'` about  how that species that started with the organ no longer has it and feels depressed.

Comment (itsmeow) [reply code/datums/mood_events/generic_negative_events.dm:99]: I think the intention is a felinid with no ears would wag, but a human with just a tail would not

Comment (Tsar-Salat) [reply code/datums/mood_events/generic_negative_events.dm:99]: > The result is that now all species can *wag, if they have a tail.

.

> but a human with just a tail would not

How did you arrive at that conclusion from what they said

**Comment by Fronsis** [comment]
I remember some people saying in discord that it makes sense for you to be unable to wag it since you just.. got stuffed a piece of flesh into your back, you don't have any nerve connection, you could add some rp-surgery way to enable that "nerve tail reconnection" (you could even apply that to broken wings) it would be more inmersive, maybe too much? Depends.

**Comment by Tsar-Salat** [comment]
> I remember some people saying in discord that it makes sense for you to be unable to wag it since you just.

If we are going that route, getting your limb replaced and regaining motor control would be an arduous process as well. 

How the hell would a human even be able to connect nerves to an ethereal or plasmaman's arm the way we can now?

I think, even from a rp perspective, we can suspend some disbelief and say these spacemen have had their bodies wired to instantly accept foreign organs and limbs through genetic modification.

Otherwise, our current limb modularization makes no sense at all.

**Comment by Fronsis** [comment]
> > I remember some people saying in discord that it makes sense for you to be unable to wag it since you just.
> 
> If we are going that route, getting your limb replaced and regaining motor control would be an arduous process as well.
> 
> How the hell would a human even be able to connect nerves to an ethereal or plasmaman's arm the way we can now?
> 
> I think, even from a rp perspective, we can suspend some disbelief and say these spacemen have had their bodies wired to instantly accept foreign organs and limbs through genetic modification.
> 
> Otherwise, our current limb modularization makes no sense at all.


I mean, as far as i remember you do take a small toxin damage whenever you get a new body part attached so.. there's that, like i said it depends how far you want to go

File: code/datums/mood_events/generic_negative_events.dm (hunk 1 above)
Comment (itsmeow) [reply code/datums/mood_events/generic_negative_events.dm:99]: This code is for tableshove wagging. Such intense waggery would require being an actual catperson and not just some tail having weirdo

Comment (Absolucy) [reply code/datums/mood_events/generic_negative_events.dm:99]: Yeah, the intent is so either true felinids or someone who's been cat surgeoned would wag from it

**Review by itsmeow — APPROVED** [review]

**Pull request merged**